src/mapping/star_align/script.py

```diff
@@ -0,0 +1,197 @@
+  'memory_pb': None,
+  'temp_dir': '/tmp'
+}
+## VIASH END
+
+########################
+### Helper functions ###
+########################
+
+# fastqgz_regex = r'([A-Za-z0-9\-_\.]+)_S(\d+)_L(\d+)_([RI]\d+)_(\d+)\.fastq(\.gz)?'
+# regex for matching R[12] fastq(gz) files
+fastqgz_regex = r'([A-Za-z0-9\-_\.]+)_S(\d+)_L(\d+)_(R\d+)_(\d+)\.fastq(\.gz)?'
+
+# helper function for cheching whether something is a gzip
+def is_gz_file(path: Path) -> bool:
+  with open(path, 'rb') as file:
+    return file.read(2) == b'\x1f\x8b'
+
+# look for fastq files in a directory
+def search_fastqs(path: Path) -> list[Path]:
+  if path.is_dir():
+    print(f"Input '{path}' is a directory, traversing to see if we can detect any FASTQ files.", flush=True)
+    value_paths = [file for file in path.iterdir() if re.match(fastqgz_regex, file.name) ]
+    return value_paths
+  else:
+    return [path]
+
+# if {par_value} is a Path, extract it to a temp_dir_path and return the resulting path
+def extract_if_need_be(par_value: Path, par_name: str, temp_dir_path: Path) -> Path:
+  print(f'>> Check compression of --{par_name} with value: {par_value}', flush=True)
+
+  if par_value.is_file() and tarfile.is_tarfile(par_value):
+    # Remove two extensions (if they exist)
+    extaction_dir_name = Path(par_value.stem).stem
+    unpacked_path = temp_dir_path / extaction_dir_name
+    print(f'  Tar detected; extracting {par_value} to {unpacked_path}', flush=True)
+
+    with tarfile.open(par_value, 'r') as open_tar:
+      members = open_tar.getmembers()
+      root_dirs = [member
+        for member in members 
+        if member.isdir() and member.name != '.' and '/' not in member.name]
+      # if there is only one root_dir (and there are files in that directory)
+      # strip that directory name from the destination folder
+      if len(root_dirs) == 1:
+        for mem in members: 
+          mem.path = Path(*Path(mem.path).parts[1:])
+      members_to_move = [mem for mem in members if mem.path != Path('.')]
+      open_tar.extractall(unpacked_path, members=members_to_move)
+    return unpacked_path
+
+  elif par_value.is_file() and is_gz_file(par_value):
+    # Remove extension (if it exists)
+    extaction_file_name = Path(par_value.stem)
+    unpacked_path = temp_dir_path / extaction_file_name
+    print(f'  Gzip detected; extracting {par_value} to {unpacked_path}', flush=True)
+    
+    with gzip.open(par_value, 'rb') as f_in:
+      with open(unpacked_path, 'wb') as f_out:
+        shutil.copyfileobj(f_in, f_out)
+    return unpacked_path
+
+  else:
+    return par_value
+
+########################
+###    Main code     ###
+########################
+
+# rename keys and convert path strings to Path
+# note: only list file arguments here. if non-file arguments also need to be renamed, 
+# the `processPar()` generator needs to be adapted
+to_rename = {'input': 'readFilesIn', 'reference': 'genomeDir', 'output': 'outFileNamePrefix'}
+
+def processPar(orig_par, to_rename):
```

Comment:
This implementation if more pythonic (although I am nitpicking here). Does this work?
```python3
def processPar(orig_par, to_rename):
  for key, value in orig_par.items():
    # rename the key in par based on the `to_rename` dict
    try:
        new_key = to_rename[key]
    except KeyError:
        yield key, value
    else:
        if isinstance(value, list):
            yield new_key, [Path(val) for val in value]
        else:
            yield new_key, Path(value)
```

Reply:
And oh, I prefer to use snakecase  for python functions :) so perhaps `process_par`?

Reply:
:+1: